backend/scripts/mirror_s3_data.sh

```diff
@@ -11,30 +11,80 @@ set -e
+    if [[ ! -z `grep 'hosted-cellxgene' <<< $bucket` ]]; then
+      rdev_bucket_suffix="cellxgene"
+    elif [[ ! -z `grep 'corpora-data' <<< $bucket` ]]; then
+      rdev_bucket_suffix="artifacts"
+    fi
+    key=$(sed -E 's/s3:\/\/([^\/]+)\/(.*)/\2/' <<< $uri)
+    [[ "${key: -1}" == "/" ]] && s3_cmd=$S3_SYNC_CMD || s3_cmd=$S3_COPY_CMD  # use 'sync' if recursive, 'cp' otherwise
+    $s3_cmd $uri s3://env-rdev-${rdev_bucket_suffix}/${STACK}/${key} || echo "$uri does not exist -- skipping..."
+  done
+
+  # Copy public assets
+  dv_id_query="select unnest(datasets) from \"CollectionVersion\" where collection_id in ('$(sed "s/,/','/g" <<< $COLLECTIONS)')"
+  dataset_version_ids=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${DEPLOYMENT_STAGE} --username corpora_${DEPLOYMENT_STAGE} --host 0.0.0.0 --csv --tuples-only -c "$dv_id_query"))
+  DEPLOYMENT_STAGE=$DEPLOYMENT_STAGE make db/tunnel/down  # db access no longer needed
+  exts=("rds" "h5ad")
+  for dv_id in "${dataset_version_ids[@]}"; do
+    for ext in "${exts[@]}"; do
+      uri="s3://dataset-assets-public-${DEPLOYMENT_STAGE}/${dv_id}.${ext}"
+      $S3_COPY_CMD $uri s3://env-rdev-datasets/${STACK}/${dv_id}.${ext} || echo "$uri does not exist -- skipping..."
+    done
+  done
+else
+  # TODO: Add --delete once we confirm that is no data in the folders
+  # that needs to be kept around (buckets are versioned, so we're not in
+  # jeopardy of losing anything permanently). This would make the
+  # operation more destructive, of course!
+  #
+  # Note: "--copy-props metadata-directive" copies s3 object metadata, but not tags.
+  # We have not granted s3:GetObjectTagging perm to dev AWS account on prod buckets,
+  # so this avoids errors. As none of the s3 objects contain tags, this is acceptable.
+
+  set -x
+  $S3_SYNC_CMD --exclude '*loom' s3://corpora-data-${SRC_ENV}/ s3://corpora-data-${DEST_ENV}/
+  $S3_SYNC_CMD s3://hosted-cellxgene-${SRC_ENV}/ s3://hosted-cellxgene-${DEST_ENV}/
+  # We need to assume the sync-datasets-{dev|staging} role, which has the necessary permissions to perform a s3 sync from
+  # {dev|staging|prod} to {dev|staging} **for the public-access datasets bucket**.
+  ACCOUNT_ID=`aws sts get-caller-identity | jq -r '.Arn' | grep -o '[0-9]*'`  # First get AWS account id for dev account
+  SYNC_ROLE_CREDENTIALS=`aws sts assume-role --role-arn arn:aws:iam::${ACCOUNT_ID}:role/sync-datasets-dev \
+                                             --role-session-name sync-datasets \
+                         | jq -r '.Credentials'`
+  # Execute sync for datasets bucket with sync role credentials
+  AWS_SESSION_TOKEN=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.SessionToken'` \
+  AWS_SECRET_ACCESS_KEY=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.SecretAccessKey'` \
+  AWS_ACCESS_KEY_ID=`echo $SYNC_ROLE_CREDENTIALS | jq -r '.AccessKeyId'` \
+  $S3_SYNC_CMD s3://dataset-assets-public-${SRC_ENV}/ s3://dataset-assets-public-${DEST_ENV}/
+  set +x
```

Comment:
assuming the else block is just a wholesale move, so not reviewed

Reply:
It has some changes: `parallel`izing the sync cmds 🥲